R/convert-from-datacamp.R

```diff
@@ -25,82 +25,65 @@ datacamp_to_learnr_pkg <-
     }
     #
     # # For accidental sourcing... -_-
     # stop()
     #
     # # Checking if project is under git and if changes are not committed
-    if (!in_repository(cur_path))
+    if (!in_repository(dc_path))
       stop("Please make that the files are under Git version control.")
-    if (length(status(cur_path)$unstaged) != 0)
+    if (length(status(dc_path)$unstaged) != 0)
       stop("Please make sure to commit changes before running.")
     #
     # # Listing all files -------------------------------------------------------
 
-    ## needs to check to see if there are dashes in the folder name
-
-    pkgname <- basename(cur_path)
-    if (grepl("-", pkgname)) {
-      if (is.null(rename)) {
-        pkgname <- gsub("-", "", pkgname)
-      } else {
-        pkgname <- rename
-      }
-
-      new_path <- strsplit(cur_path, "/")[[1]]
-      new_path <- new_path[-length(new_path)]
-      # new_path <- new_path[-1]
-      new_path <- paste(new_path, collapse = .Platform$file.sep)
-      new_path <- file.path(new_path, pkgname)
-      file.rename(cur_path, new_path)
-      cur_path <- new_path
-    }
+
+    pkgname <- basename(lrnr_pkg_path)
     pkgname <- paste(pkgname, ".Rproj", sep = "")
 
     all_files <-
-      fs::dir_ls(cur_path,
+      fs::dir_ls(dc_path,
              recurse = TRUE,
              type = "file",
              all = TRUE)
-    proj_name <- str_subset(all_files, "\\.Rproj$")
 
 
 
-    #### No RPoject, which would seem to be default??
-    if (length(proj_name) == 0) {
-      tmp <- paste(pkgname, ".Rproj", sep = "")
-      proj_name = file.path(cur_path, tmp)
-    }
     # # Create as package -------------------------------------------------------
-
-    usethis::create_package(fs::path_dir(proj_name))
+    ### this will fail if there are dashes, in the name, could provide an earlier error
+    usethis::create_package(lrnr_pkg_path)
 
     chapters <- str_subset(all_files, "chapter.\\.md")
 
     ### Gets chapter names
     chapters_name <- fs::path_ext_remove(fs::path_file(chapters))
 
+
     ## creates demo files
-    sapply(chapters_name,
-           usethis::use_tutorial,
-           open = F,
-           title = "Demo")
+    withr::with_dir(lrnr_pkg_path, {
+      sapply(chapters_name,
+             usethis::use_tutorial,
+             open = F,
+             title = "Demo")
+    } )
+
 
     ## Overwrites demo files
     new_chapter_path <-
-      file.path(cur_path, "inst/tutorials/", fs::path_file(chapters)) %>%
+      file.path(lrnr_pkg_path, "inst/tutorials/", fs::path_file(chapters)) %>%
       str_replace("\\.md", ".Rmd")
     fs::file_move(chapters, new_chapter_path)
 
-    ### will need to be done within right directory
-    usethis::use_build_ignore("slides")
-    usethis::use_tidy_versions()
-    usethis::use_github(overwrite = T)
-    usethis::use_ccby_license(name = author_name)
-    usethis::use_blank_slate()
+    withr::with_dir(lrnr_pkg_path, {
+      usethis::use_build_ignore("slides")
```

Comment:
This build ignore might be conditional. Depends on if the slides are slides or if they are just text. I haven't created videos or literal slides for my slide files, so I was planning on simply converting them to text and inserting into the chapter file. So, this could be ignored or may be needed. For now let's comment this out ok?